From da27cb1923dd0bcda2afd5afbbee2cb11d8d1ff9 Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tduszynski@cavium.com>
Date: Wed, 13 Mar 2019 16:32:00 +0100
Subject: [PATCH 042/386] octeontx2-af: fix missing function prototype warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

cgx_lmac_pasue_frm_config() does not need to be exported and can be
marked static because noone else seems to be using it.

drivers/net/ethernet/marvell/octeontx2/af/cgx.c:476:6: warning: no previous
prototype for ‘cgx_lmac_pause_frm_config’ [-Wmissing-prototypes]
 void cgx_lmac_pause_frm_config(struct cgx *cgx, int lmac_id, bool enable)
      ^~~~~~~~~~~~~~~~~~~~~~~~~

Change-Id: Id5bb110a03f5d93dd9338cc5b7eeaee3ea8fbab7
Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5675
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 3 +--
 drivers/net/ethernet/marvell/octeontx2/af/cgx.h | 1 +
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 081fed52add6..33453bc1abfe 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -460,7 +460,7 @@ int cgx_lmac_set_pause_frm(void *cgxd, int lmac_id,
 }
 EXPORT_SYMBOL(cgx_lmac_set_pause_frm);
 
-void cgx_lmac_pause_frm_config(struct cgx *cgx, int lmac_id, bool enable)
+static void cgx_lmac_pause_frm_config(struct cgx *cgx, int lmac_id, bool enable)
 {
 	u64 cfg;
 
@@ -514,7 +514,6 @@ void cgx_lmac_pause_frm_config(struct cgx *cgx, int lmac_id, bool enable)
 		cgx_write(cgx, lmac_id, CGXX_SMUX_TX_CTL, cfg);
 	}
 }
-EXPORT_SYMBOL(cgx_lmac_pause_frm_config);
 
 /* CGX Firmware interface low level support */
 static int cgx_fwi_cmd_send(u64 req, u64 *resp, struct lmac *lmac)
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.h b/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
index 0d4368bf3cd5..6e4882075e1b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
@@ -140,4 +140,5 @@ int cgx_lmac_get_pause_frm(void *cgxd, int lmac_id,
 			   u8 *tx_pause, u8 *rx_pause);
 int cgx_lmac_set_pause_frm(void *cgxd, int lmac_id,
 			   u8 tx_pause, u8 rx_pause);
+
 #endif /* CGX_H */
-- 
2.17.1

